launch/docker.go

```diff
@@ -89,16 +91,21 @@ func (d *docker) runBuild(buildConfig buildConfig) error {
 		dockerCommandOptions = append([]string{fmt.Sprintf("-m%s", buildConfig.MemoryLimit)}, dockerCommandOptions...)
 	}
 
-	err = execDockerCommand(append(dockerCommandArgs, dockerCommandOptions...)...)
+	err = d.execDockerCommand(append(dockerCommandArgs, dockerCommandOptions...)...)
 	if err != nil {
 		return fmt.Errorf("failed to run build container: %v", err)
 	}
 
 	return nil
 }
 
-func execDockerCommand(args ...string) error {
-	cmd := execCommand("docker", args...)
+func (d *docker) execDockerCommand(args ...string) error {
+	cmd := (*exec.Cmd)(nil)
+	if d.useSudo {
+		cmd = execCommand("sudo", append([]string{"docker"}, args...)...)
+	} else {
+		cmd = execCommand("docker", args...)
+	}
```

Comment:
Is it better to write the following?
```suggestion
	commands := append([]string{"docker"}, args...)
	if d.useSudo {
		commands = append([]string{"sudo"}, commands...)
	}
	cmd := execCommand(commands[0], commands[1:]...)
```

Reply:
thx suggestion.